docs: include a section with CircleCI instructions (#318)
game-ci/documentation: docs/11-circleci/05-executors.mdx

> ---
> toc_max_heading_level: 4
> ---

Reviewer: ```suggestion
```

Reviewer: The 4th level doesn't really add clarity for this page in my opinion. It actually becomes harder to find your executor  from the list like this. Recommend leaving it default.

# Executors

> The orb provides reusable execution environments for you to run your jobs on. The executors
> available are [macOS](https://circleci.com/developer/orbs/orb/game-ci/unity#executors-macos),
> [macOS on Runner](https://circleci.com/developer/orbs/orb/game-ci/unity#executors-macos-runner),
> [Windows](https://circleci.com/developer/orbs/orb/game-ci/unity#executors-windows-2022),
> [Windows on Runner](https://circleci.com/developer/orbs/orb/game-ci/unity#executors-windows-runner)
> and [Ubuntu on Docker](https://circleci.com/developer/orbs/orb/game-ci/unity#executors-ubuntu).

Reviewer: ```suggestion
The Orb provides reusable execution environments for you to run your jobs on. The executors
available are [macOS](https://circleci.com/developer/orbs/orb/game-ci/unity#executors-macos),
[macOS on Runner](https://circleci.com/developer/orbs/orb/game-ci/unity#executors-macos-runner),
[Windows](https://circleci.com/developer/orbs/orb/game-ci/unity#executors-windows-2022),
[Windows on Runner](https://circleci.com/developer/orbs/orb/game-ci/unity#executors-windows-runner)
and [Ubuntu on Docker](https://circleci.com/developer/orbs/orb/game-ci/unity#executors-ubuntu).
```

Reviewer: Perhaps we can use an unordered list here.

Author: I tried the unordered list, but it doesn't look super good. Lots of white space. Since the quick access component on the right side already has a list, I'll keep it as it is for now.

> Due to the installation of external dependencies, the `macos` and `windows` executors have longer
> environment preparation times. This can be mitigated using their
> [Runner](https://circleci.com/docs/runner-overview) variants. As Runner doesn't share the ephemeral
> nature of the CircleCI cloud executors, the external dependencies will already be installed after
> the first run, decreasing the environment preparation time.

Reviewer: ```suggestion
The `macos` and `windows` executors take longer to setup due to the installation of external dependencies. 
This can be mitigated using their
[Runner](https://circleci.com/docs/runner-overview) variants. As Runner doesn't share the ephemeral
nature of the CircleCI cloud executors, the external dependencies will already be installed after
the first run, decreasing the environment preparation time.
```

[Installing Self-Hosted Runners with the Web App](https://circleci.com/docs/runner-installation) and
[CircleCI Self-hosted Runner Installation on Windows](https://circleci.com/docs/runner-installation-windows).

The `windows-runner` is an excellent alternative over [windows](#windows) to build Windows IL2CPP
with faster build times. Its non-ephemeral nature saves times on installing external dependencies
after the first run. And in addition, you are free to use an agent with hardware exceeding the
[available](https://circleci.com/docs/configuration-reference#windows-execution) for the web Windows
executor.

#### editor_version

- _Default_: `N/A`
- _Required_: `True`
- _Type_: `String`

Choose the Unity Editor version that you want to use to build your project. The available options
can be found on the [versions](/docs/docker/versions) page.

```yaml
version: 2.1

orbs:
  unity: game-ci/unity@x.y

workflows:
  build-unity-project:
    jobs:
      - unity/build:
          executor:
            name: 'unity/windows-runner'
            editor_version: '2021.3.2f1'
```

#### resource_class

- _Default_: `N/A`
- _Required_: `True`
- _Type_: `String`

> Enter your [runner resource class](https://circleci.com/docs/runner-overview). If you are unsure
> what this means, navigate to the beginning of this [sub-section](#windows-runner).

Reviewer: ```suggestion
Enter your [runner resource class](https://circleci.com/docs/runner-overview). If you are unsure
what this means, navigate to the beginning of [this section](#windows-runner).
```

```yaml
version: 2.1

orbs:
  unity: game-ci/unity@x.y

workflows:
  build-unity-project:
    jobs:
      - unity/build:
          executor:
            name: 'unity/windows-runner'
            resource_class: 'my_runner_resource_class'
```

#### working_directory

- _Default_: `C:\Users\circleci\AppData\Local\Temp\Workdir`
- _Required_: `False`
- _Type_: `String`

> Enter the working directory of your runner agent. This will be where the job steps will run.

Reviewer: ```suggestion
Enter the working directory of your runner agent. This is be where the job steps will run from.
```
